inst/tests/tests.Rraw

```diff
@@ -9898,6 +9898,10 @@ unlink(f)
 test(1761.1, fread("1\n2\n3", fill=TRUE), data.table(V1=1:3))
 test(1761.2, fread("1\n2\n3", fill=FALSE), data.table(V1=1:3))
 
+# CJ should retain timezone information, #2029
+df <- CJ(week=as.POSIXct('2016-01-01', tz = 'UTC'), id=1:10)
+test(1762, attr(df$week, 'tzone'), 'UTC')
```

Comment:
More tests please including multi-column data.table with 1 and 2 columns of POSIXct with other non-POSIXct (say plain integer) columns at the beginning, middle and end.  Enough variations to test that the code is actually retaining the tzone on the desired columns and not adding tzones to columns that shouldn't, because I'm not sure it's right currently just by glancing at it.  The test is just a single column.